src/main/java/olya/app/remindme/controller/UserController.java

```diff
@@ -22,25 +28,74 @@
 public class UserController {
     private final UserService userService;
     private final UserMapper userMapper;
+    private final TokenService tokenService;
 
+    //public
     @PostMapping("/register")
     public UserResponseDto register(@Valid @RequestBody UserRequestDto userRequestDto) {
         User user = userService.create(userRequestDto);
         return userMapper.mapToDto(user);
     }
 
+    //for user and admin
+    @GetMapping("/my-user")
+    public UserResponseDto getCurrentUser(@AuthenticationPrincipal UserPrincipal principal) {
+        User user = userService.getByEmail(principal.getEmail());
+        return userMapper.mapToDto(user);
+    }
+
+    //for user and admin
+    @PutMapping("/my-user")
+    public UserResponseDto updateCurrentUser(@AuthenticationPrincipal UserPrincipal principal, @Valid @RequestBody UserRequestDto userRequestDto) {
+        User user = userService.getByEmail(principal.getEmail());
+        User updatedUser = userService.update(user, userRequestDto);
+        return userMapper.mapToDto(updatedUser);
+    }
+
+    //for user and admin
+    @PutMapping("/my-user/deactivate")
+    public void deactivateCurrentUser(@AuthenticationPrincipal UserPrincipal principal) {
+        User user = userService.getByEmail(principal.getEmail());
+        userService.deactivate(user);
+    }
+
+    //for admin only
+    //get user by email
+    @GetMapping("/by-email")
+    public User getByEmail(@RequestParam String email) {
+        return userService.getByEmail(email);
+    }
+
+    //for admin only
+    @GetMapping("/all")
+    public List<User> getAllUsers() {
+       return userService.getAll();
+    }
+
+    //for admin only
     @PutMapping("/{id}")
-    public UserResponseDto update(@PathVariable Long id, @RequestBody UserRequestDto userRequestDto) {
-        return null;
+    public User update(@PathVariable Long id, @Valid @RequestBody UserRequestDto userRequestDto) {
+        User user = userService.getById(id);
+        return userService.update(user, userRequestDto);
     }
 
+    //for admin only
     @PutMapping("/deactivate/{id}")
     public void deactivate(@PathVariable Long id) {
+        User user = userService.getById(id);
+        userService.deactivate(user);
+    }
 
+    //for admin only
+    @PutMapping("/change-roles/{id}")
+    public void updateRoles(@PathVariable Long id, @RequestBody Set<Role.RoleName> newRoles) {
+        User user = userService.getById(id);
+        userService.updateRoles(user, newRoles);
     }
 
-    @GetMapping("/by-email")
-    public UserResponseDto getByEmail(@RequestParam String email) {
-        return null;
+    @PutMapping("/revoke-tokens/{id}")
```

Comment:
Try to use resource based naming for endpoints. In this case it would be `/tokens/{id}/revocation`